soda/contracts/README.md

```diff
@@ -0,0 +1,107 @@
+# Goal and purpose of data contracts in Soda
+
+Soda's goal is to be the best enforcement engine for data contract implementations. For
+this purpose we have created a new YAML format that aligns with data contracts from a producer
+perspective.  This new format enables a very easy translation from any data contract
+YAML language to Soda's enforcable contract language.
+
+# Example
+
+```yaml
+name: DIM_CUSTOMER
+
+columns:
+  - name: id
+    data_type: character varying
+    unique: true
+  - name: cst_size
+    data_type: decimal
+  - name: cst_size_txt
+    valid_values: [1, 2, 3]
+  - name: distance
+    data_type: integer
+  - name: country
+    data_type: varchar
+    not_null: true
+  - name: ts
+
+checks:
+  - avg(distance) between 400 and 500
+```
+
+# Top level keys
+
+| Key | Description | Required | YAML data type |
+| --- | ----------- | -------- | -------------- |
+| `name` | Name of the dataset as in the SQL engine. | Required | string |
```

Comment:
I would advise for a more specific name here. Judging by your description this is the dataset name. I would therefore expect the key to be called `dataset` no?

Reply:
Good point as I was in doubt.  I first called it `dataset` as you suggest.  

i wanted to make it consistent with the columns. In the context of a column, it's normal to call it `name`.  Not sure if that needs consistency.  Also realized that all the properties on file level are related to the dataset.  I also considered `dataset_name`.  

Other option is to use `physical_name` in case we also want non-tech names in there later.

Reply:
At this point and given unsure directions of what to add and so on, I'd advocate for the most specific name for now `dataset_name`. Then later if we want to let other things in then we'll likely not clash and I'm sure that if we want to introduce abstractions like `phsycal_name` or whatever we can find a way to reason about defaults and so on.

What do you think?